File: exo20/exo20.py
import time

def now():
    return int(time.time() * 1000)

def get_time_exec(callback):
    begin = now()
    print("get_time_exec, begin:", begin)
    callback()
    end = now()
    print("get_time_exec, end:", end)
    return f"{(end - begin) / 1000} s écoulées"

def add(a, b):
    return a + b

def add_arr(arr):
    return sum(arr)

def facto(n):
    if n == 0:
        return 1
    return n * facto(n - 1)

print(get_time_exec(lambda: facto(500)))

File: exo20/exo20.js
const now = () => {
    const rightNow = new Date();
    return rightNow.getTime();
};

const getTimeExec = (callback) => {
    const begin = now();
    console.log("getTimeExec , begin:", begin);
    callback();
    const end = now();
    console.log("getTimeExec , end:", end);
    return `${(end - begin) / 1000} s écoulées`;
};

const add = (a, b) => {
    return a + b;
};
const addArr = (arr) => {
    return arr.reduce((acc, curr) => acc + curr, 0);
};

const facto = (n) => {
    if (n === 0) return 1;
    return n * facto(n - 1);
};

console.log(getTimeExec(() => facto(500)));
